src/App.js

```diff
@@ -1,36 +1,188 @@
-import './style/App.css';
-import { Route, Routes, useLocation} from 'react-router-dom';
+import "./style/App.css";
 
+import React, { useState, useEffect } from "react";
+
+import axios from "axios";
+import { Modal, Button, Form, Card } from "react-bootstrap";
+import "bootstrap/dist/css/bootstrap.css";
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { fas } from "@fortawesome/free-solid-svg-icons";
+library.add(fas);
 
 function App() {
-  const location = useLocation();
-  const background = location.state && location.state.background;
+
+  const [show, setShow] = useState(false);
+
+  const handleClose = () => setShow(false);
+  const handleShow = () => setShow(true);
+
+  const [addNewTaskTitle, setaddNewTaskTitle] = useState("");
+  const [taskDescription, settaskDescription] = useState("");
+  const [dueDate, setDueDate] = useState(new Date());
+
+  const postData = () => {
+    axios
+      .post(`https://64ef1ed7219b3e2873c3f9ad.mockapi.io/todoData`, {
+        addNewTaskTitle,
+        taskDescription,
+        dueDate,
+      })
+      .then(() => {
+        getData();
+      });
+  };
+
+  //// from read
+
+  const [APIData, setAPIData] = useState([]);
+  useEffect(() => {
+    axios
+      .get(`https://64ef1ed7219b3e2873c3f9ad.mockapi.io/todoData`)
+      .then((response) => {
+        setAPIData(response.data);
+      });
+  }, []);
+
+  const setData = (data) => {
+    console.log(data);
+    let { id, addNewTaskTitle, taskDescription, dueDate } = data;
+    localStorage.setItem("ID", id);
+    localStorage.setItem("Your Task", addNewTaskTitle);
+    localStorage.setItem("Due Date", dueDate);
+    localStorage.setItem("To-Do", taskDescription);
+  };
+
+  const onDelete = (id) => {
+    axios
+      .delete(`https://64ef1ed7219b3e2873c3f9ad.mockapi.io/todoData/${id}`)
+      .then(() => {
+        getData();
+      });
+  };
+  const getData = () => {
+    axios
+      .get(`https://64ef1ed7219b3e2873c3f9ad.mockapi.io/todoData`)
+      .then((getData) => {
+        setAPIData(getData.data);
+      });
+  };
+
 
   return (
-      <div className="App">
-        <div className='todo-menu'>
-            <p>
-              To-DO List
-            </p>
-            <Routes location={background || location}>
-              <Route path="/" element={<AddTaskButton />}>
-              <Route path='/read' element={<Read />} />
-                <Route path="modal" element={<Create />} />
-              </Route>
-              {/* <Route path='/read' element={<Read />} /> */}
-              <Route path='/update' element={<Update />} />
-              </Routes>
-                {background && (
-              <Routes>
-                <Route path="modal" element={<Create />} />
-              </Routes>
-      )}
-          </div>
+    <div className="App">
```

Comment:
Well, the App page is quite large. I propose to divide it into components.

I suggest the following structure:
PageLayout is base component. There are title of page and children items.
- TodoLIst is a component that displays a to-do list.
-- Todo, there will be the creation date, title, description and buttons
--- Button a component that accepts an icon and a click event

Reply:
Done! I changed the structure and split the App.js page into the components as you wrote.